physical/dynamodb/dynamodb.go

```diff
@@ -155,16 +155,14 @@ func NewDynamoDBBackend(conf map[string]string, logger log.Logger) (physical.Bac
 		writeCapacity = DefaultDynamoDBWriteCapacity
 	}
 
-	accessKey := os.Getenv("AWS_ACCESS_KEY_ID")
-	if accessKey == "" {
+	var accessKey, secretKey, sessionToken string
+	if os.Getenv("AWS_ACCESS_KEY_ID") == "" {
 		accessKey = conf["access_key"]
 	}
-	secretKey := os.Getenv("AWS_SECRET_ACCESS_KEY")
-	if secretKey == "" {
+	if os.Getenv("AWS_SECRET_ACCESS_KEY") == "" {
 		secretKey = conf["secret_key"]
 	}
-	sessionToken := os.Getenv("AWS_SESSION_TOKEN")
-	if sessionToken == "" {
+	if os.Getenv("AWS_SESSION_TOKEN") == "" {
```

Comment:
We should just drop pulling all three of these here.

Reply:
The point of keeping it there was so that if you overrode the config with env vars we'd prioritize the env vars -- note that awsutil has the ordering static, env, shared, iam. Whether this is the right thing to do or not is debatable, but it's also how it's worked up to this point so I'm not sure changing them is the right move.

Reply:
Oh I see! Maybe it would be nice to add a comment to that effect, then. But yes, I looked through the code in `GenerateCredentialChain` and I see why you're doing that but still using the method, that method does a lot of work.